lib/rules/no-irregular-whitespace/index.mjs

```diff
@@ -61,48 +75,65 @@ const rule = (primary) => {
 		}
 
 		/**
-		 * @param {import('postcss').Node} node
+		 * @template {import('postcss').Node} T
+		 * @param {T} node
 		 * @param {string | undefined} value
+		 * @param {(node: T) => number} getIndex
 		 */
-		const validate = (node, value) => {
-			const issue = value && findIrregularWhitespace(value);
+		const validate = (node, value, getIndex) => {
+			if (!value) return;
+
+			const issues = findIrregularWhitespace(value);
+
+			if (!issues.length) return;
 
-			if (issue) {
+			const startIndex = getIndex(node);
+
+			issues.forEach(({ index, length }) => {
 				report({
 					ruleName,
 					result,
 					message: messages.unexpected,
 					node,
-					word: issue,
+					index: startIndex + index,
+					endIndex: startIndex + index + length,
 				});
-			}
+			});
 		};
 
 		root.walkAtRules((atRule) => {
-			validate(atRule, atRule.name);
-			validate(atRule, atRule.params);
-			validate(atRule, atRule.raws.before);
-			validate(atRule, atRule.raws.after);
-			validate(atRule, atRule.raws.afterName);
-			validate(atRule, atRule.raws.between);
+			validate(atRule, atRule.raws.before, zeroIndex);
+			validate(atRule, atRule.name, oneIndex);
+			validate(atRule, atRule.raws.afterName, atRuleRawsAfterNameIndex);
+			validate(atRule, getAtRuleParams(atRule), atRuleParamIndex);
+			validate(atRule, atRule.raws.between, atRuleRawsBetweenIndex);
+			validate(atRule, atRule.raws.after, atRuleRawsAfterIndex);
 		});
 
 		root.walkRules((ruleNode) => {
-			validate(ruleNode, ruleNode.selector);
-			validate(ruleNode, ruleNode.raws.before);
-			validate(ruleNode, ruleNode.raws.after);
-			validate(ruleNode, ruleNode.raws.between);
+			validate(ruleNode, ruleNode.raws.before, zeroIndex);
+			validate(ruleNode, getRuleSelector(ruleNode), zeroIndex);
+			validate(ruleNode, ruleNode.raws.between, ruleRawsBetweenIndex);
+			validate(ruleNode, ruleNode.raws.after, ruleRawsAfterIndex);
 		});
 
 		root.walkDecls((decl) => {
-			validate(decl, decl.prop);
-			validate(decl, decl.value);
-			validate(decl, decl.raws.before);
-			validate(decl, decl.raws.between);
+			validate(decl, decl.raws.before, zeroIndex);
+			validate(decl, decl.prop, zeroIndex);
+			validate(decl, decl.raws.between, declarationRawsBetweenIndex);
+			validate(decl, getDeclarationValue(decl), declarationValueIndex);
 		});
 	};
 };
 
+function zeroIndex() {
+	return 0;
+}
+
+function oneIndex() {
+	return 1;
+}
+
```

Comment:
```suggestion

/**
 * @param {import('postcss').Rule} rule
 * @returns {number}
 */
function ruleRawsBetweenIndex(rule) {
	return rule.selector.length;
}

```

What about putting it here instead?
It's unlikely it will be reused and even if it was it's too simple to warrant being an utility IMHO.